pkg/host-disk/disk.go

```diff
@@ -0,0 +1,63 @@
+package hostdisk
+
+import (
+	"fmt"
+	"path/filepath"
+
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/record"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/log"
+
+	"kubevirt.io/kubevirt/pkg/util"
+)
+
+type PVCDiskImgCreator struct {
+	diskImgCreator diskImgCreator
+}
+
+func NewPVCDiskImgCreator(recorder record.EventRecorder, lessPVCSpaceToleration int, minimumPVCReserveBytes uint64) PVCDiskImgCreator {
+	return PVCDiskImgCreator{
+		diskImgCreator: newDiskImgCreator(recorder, lessPVCSpaceToleration, minimumPVCReserveBytes),
+	}
+}
+
+func getPVCInfo(vmi *v1.VirtualMachineInstance, volumeName string) (*v1.PersistentVolumeClaimInfo, error) {
+	for _, volumeStatus := range vmi.Status.VolumeStatus {
+		if volumeStatus.Name == volumeName {
+			if volumeStatus.PersistentVolumeClaimInfo.VolumeMode != nil && *volumeStatus.PersistentVolumeClaimInfo.VolumeMode != k8sv1.PersistentVolumeFilesystem {
+				return nil, fmt.Errorf("volume %s is not in filesystem mode", volumeName)
+			}
+			return volumeStatus.PersistentVolumeClaimInfo, nil
+		}
+	}
+
+	return nil, fmt.Errorf("no disk found")
+}
+
+func (c PVCDiskImgCreator) Create(vmi *v1.VirtualMachineInstance, volumeName, diskPath string) error {
+	pvcInfo, err := getPVCInfo(vmi, volumeName)
+	if err != nil {
+		return err
+	}
+
+	capacity, capacityOk := pvcInfo.Capacity[k8sv1.ResourceStorage]
+	requested, requestedOk := pvcInfo.Requests[k8sv1.ResourceStorage]
+
+	if !capacityOk && !requestedOk {
+		return fmt.Errorf("unable to determine capacity from PVC that provides no storage capacity or requests")
+	}
+	var requestedSize int64
+	if requestedOk && ((capacityOk && capacity.Value() > requested.Value()) || !capacityOk) {
```

Comment:
Just a simplification...
```suggestion
	if requestedOk && (!capacityOk || capacity.Value() > requested.Value()) {
```